oshmem/mca/memheap/base/memheap_base_frame.c

```diff
@@ -36,6 +36,7 @@ int mca_memheap_base_key_exchange = 1;
 opal_list_t mca_memheap_base_components_opened = {{0}};
 int mca_memheap_base_already_opened = 0;
 mca_memheap_map_t mca_memheap_base_map = {{{{0}}}};
+int mca_memheap_base_max_segments = MCA_MEMHEAP_MAX_SEGMENTS;
```

Comment:
why need to limit it by default to 32?

Reply:
> why need to limit it by default to 32?

used as allowed number of segments to allocate, when limit is reached - warning prompted